python/cudf/cudf/core/frame.py

```diff
@@ -573,6 +574,40 @@ def equals(self, other, **kwargs):
         else:
             return self._index.equals(other._index)
 
+    def _explode(self, explode_column: Any, ignore_index: bool):
+        """Helper function for `explode` in Series and Dataframe.
+        if the designated column to explode is non-nested, a copy
+        of the frame is returned. Otherwise, if ignore_index is
+        set, the original index is not exploded and will use
+        a `RangeIndex`.
+        """
+        if (
+            isinstance(self, (cudf.Series, cudf.DataFrame))
+            and self._index is not None
+        ):
+            if not is_list_dtype(self._data[explode_column].dtype):
+                data = self._data.copy(deep=True)
+                idx = None if ignore_index else self._index.copy(deep=True)
+                return self.__class__._from_data(data, index=idx)
+
+            explode_column_num = self._column_names.index(explode_column)
+            if not ignore_index:
+                explode_column_num += self._index.nlevels
+
+            res_tbl = libcudf.lists.explode_outer(
+                self, explode_column_num, ignore_index
+            )
+            res = self.__class__._from_table(res_tbl)
+
+            res._data.multiindex = self._data.multiindex
+            res._data._level_names = self._data._level_names
+
+            if not ignore_index:
+                res.index.names = self._index.names
+            return res
+        else:
+            raise NotImplementedError("_explode is not implemented for Index.")
```

Comment:
Is there a reason to artificially block this? It's okay if Pandas doesn't support it.

Reply:
Nope - this branch is more like a stub from the above. Will remove.